test/test_extended_models.py

```diff
@@ -73,10 +74,24 @@ def test_get_model_weights(name, weight):
     ],
 )
 def test_weights_copyable(copy_fn, name):
-    model_weights = models.get_model_weights(name)
-    for weights in list(model_weights):
-        copied_weights = copy_fn(weights)
-        assert copied_weights is weights
+    for weights in list(models.get_model_weights(name)):
+        assert copy_fn(weights) == weights
```

Comment:
> Isn't `copy_fn(weights) is weights` still True?

Yes.

> Maybe we should still keep it in the tests, and acknowledge in a comment that this is a somewhat surprising (but builtin) behaviour of enum fields?

I think we should ask whether we want to enforce it or not. I'm ok doing this to align with the builtin enums, but I think both an equality or identity check is fine. Your choice.

In any case, we should use the same for the serialization test.

Reply:
I'd be in favour of adding the `is` check in addition to the `==` check. Not because this is something we want to enforce (I don't think we should, hence the suggested comments), but mostly as documentation in case similar issues pop-up in the future.